src/main/java/pl/joegreen/sergeants/simulator/Simulator.java

```diff
@@ -0,0 +1,114 @@
+package pl.joegreen.sergeants.simulator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.joegreen.sergeants.api.response.GameStartApiResponse;
+import pl.joegreen.sergeants.api.response.GameUpdateApiResponse;
+import pl.joegreen.sergeants.framework.model.GameResult;
+import pl.joegreen.sergeants.framework.model.api.GameStartedApiResponseImpl;
+import pl.joegreen.sergeants.framework.model.api.UpdatableGameState;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+
+/**
+ * An offline local server simulator of one generals.io game.
+ * Single threaded, so all action calls must be made from the same thread.
+ * Use SimulatorFactory to create an instance.
+ */
+public class Simulator {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(Simulator.class);
+    private final int maxTurns;
+    private final Player players[];
+    private final GameMap gameMap;
+
+    Simulator(GameMap gameMap, int maxTurns, Player[] players) {
+        this.gameMap = gameMap;
+        this.maxTurns = maxTurns;
+        this.players = players;
+    }
+
+    /**
+     * Start simulation
+     *
+     * @return the winner
+     */
+    public Optional<Player> start() {
+        LOGGER.info("Starting match with: {}", Arrays.toString(players));
+        Arrays.stream(players).forEach(p -> {
+            GameStartApiResponse startData = getStartData(p.getPlayerIndex());
+            p.getBot().onGameStarted(new GameStartedApiResponseImpl(startData));
+        });
+
+        Arrays.stream(players).forEach(this::sendInitialGameUpdate);
+
+        for (; ; ) {
+            Arrays.stream(players)
```

Comment:
My idea is that simulator will be used to test different settings of a bot. Thus I think the caller should decide starting order.